src/ja/nicovideoseiga/src/eu/kanade/tachiyomi/extension/ja/nicovideoseiga/NicovideoSeiga.kt

```diff
@@ -1,216 +1,183 @@
-                    // A larger thumbnail is only available after going into the details page
-                    thumbnail_url = mangaElement.select(".comic_icon > div > a > img").attr("src")
-                    val statusText =
-                        mangaElement.select(".mg_description_header > .mg_icon > .content_status > span")
-                            .text()
-                    status = when (statusText) {
-                        "連載" -> {
-                            SManga.ONGOING
-                        }
-                        "完結" -> {
-                            SManga.COMPLETED
-                        }
-                        else -> {
-                            SManga.UNKNOWN
-                        }
-                    }
+                    title = manga.title
+                    author = manga.author
+                    // The thumbnail provided only displays a glimpse of the latest chapter. Not the actual cover
+                    // We can obtain a better thumbnail when the user clicks into the details
+                    thumbnail_url = manga.thumbnailUrl
+                    setUrlWithoutDomain("$baseUrl/comic/${manga.id}")
                 },
             )
         }
-        return MangasPage(mangas, mangaCount - mangaPerPage * currentPage > 0)
+        // The api call allows a maximum of 5 pages
+        return MangasPage(mangas, pageNumber < 5)
     }
 
-    override fun latestUpdatesRequest(page: Int): Request =
-        GET("$baseUrl/manga/list?page=$page&sort=manga_updated")
-
-    override fun popularMangaParse(response: Response): MangasPage = latestUpdatesParse(response)
-
     override fun popularMangaRequest(page: Int): Request =
-        GET("$baseUrl/manga/list?page=$page&sort=manga_view")
+        // This is the only API call that doesn't use the API url
+        GET("$baseUrl/manga/ajax/ranking?span=total&category=all&page=$page")
+
+    // Parses the common manga entry object from the api
+    private fun parseMangaEntry(entry: Manga): SManga {
+        // The description is html. Simply using Jsoup to remove all the html tags
+        val descriptionText = Jsoup.parse(entry.meta.description).wholeText()
+        return SManga.create().apply {
+            title = entry.meta.title
+            description = descriptionText
+            // Although their API does contain individual author fields, they are arbitrary strings and we can't trust it conforms to a format
+            // Use display name instead which puts all of the people involved together
+            author = entry.meta.author
+            thumbnail_url = entry.meta.thumbnailUrl
+            // Use the share URL so that WebView still works
+            setUrlWithoutDomain(entry.meta.shareUrl)
+            status = when (entry.meta.serialStatus) {
+                "serial" -> SManga.ONGOING
+                "concluded" -> SManga.COMPLETED
+                else -> SManga.UNKNOWN
+            }
```

Comment:
```suggestion
            }
            initialized = true
```
to avoid calling for details again, since all the necessary information is already available